test-d/get.ts

```diff
@@ -0,0 +1,50 @@
+import {Get} from '../ts41/get';
+import {expectType} from 'tsd';
+
+interface ApiResponse {
+  hits: {
+    hits: Array<{
+      _id: string;
+      _source: {
+        name: Array<{
+          given: string[];
+          family: string;
+        }>;
+        birthDate: string;
+      };
+    }>;
+  };
+}
+
+expectType<Get<ApiResponse, 'hits.hits[0]._source.name'>>([{given: ['Homer', 'J'], family: 'Simpson'}]);
+expectType<Get<ApiResponse, 'hits.hits.0._source.name'>>([{given: ['Homer', 'J'], family: 'Simpson'}]);
+expectType<Get<ApiResponse, 'hits.hits[12345]._source.name'>>([{given: ['Homer', 'J'], family: 'Simpson'}]);
+
+expectType<Get<ApiResponse, 'hits.someNonsense.notTheRightPath'>>({} as never);
```

Comment:
Please let me know if there's a better way to ensure `never` from `expectType`, or to do type-to-type (rather than type-to-value) comparisons.

I have a similar library called [expect-type](https://npmjs.com/package/expect-type) which I'd normally use, the syntax in this case would be a bit clearer (I find the difference between actual and expected here confusing) and wouldn't require the dummy variable:

```typescript
expectTypeOf<Get<ApiResponse, 'hit.hits[0]._source.name'>>()
  .toEqualTypeOf<Array<{ given: string[]; family: string }>>()

expectTypeOf<Get<ApiResponse, 'hits.someNonsense.notTheRightPath'>>().toBeNever()
```

Reply:
This is actually a bug with the tool we use: https://github.com/SamVerschueren/tsd/issues/78 Feel free to bring in your package here.